### ISSUE TITLE
Figure out why the javascript binary sizes are so large

### ISSUE THREAD

**Issue by rachelrosengoogle** [conversationai/perspectiveapi-authorship-demo#50]
Current binary sizes of code compiled with yarn build:prod :

Uncompressed:
-rw-r--r-- 1 rachelrosen primarygroup         5.7k 2018/04/18 15:14:03 build/static/inline.bundle.js
-rw-r--r-- 1 rachelrosen primarygroup         161k 2018/04/18 15:14:03 build/static/main.bundle.js
-rw-r--r-- 1 rachelrosen primarygroup         295k 2018/04/18 15:14:03 build/static/polyfills.bundle.js
-rw-r--r-- 1 rachelrosen primarygroup          57k 2018/04/18 15:14:03 build/static/styles.bundle.js
-rw-r--r-- 1 rachelrosen primarygroup         7.0M 2018/04/18 15:14:03 build/static/vendor.bundle.js

= 7.52 M

Using gzip:
-rw-r--r-- 1 rachelrosen primarygroup         1.6k 2018/04/18 15:14:03 build/static/inline.bundle.js.gz
-rw-r--r-- 1 rachelrosen primarygroup          28k 2018/04/18 15:14:03 build/static/main.bundle.js.gz
-rw-r--r-- 1 rachelrosen primarygroup          62k 2018/04/18 15:14:03 build/static/polyfills.bundle.js.gz
-rw-r--r-- 1 rachelrosen primarygroup          10k 2018/04/18 15:14:03 build/static/styles.bundle.js.gz
-rw-r--r-- 1 rachelrosen primarygroup         1.2M 2018/04/18 15:14:03 build/static/vendor.bundle.js.gz

= 1.3M

**Comment by rachelrosengoogle** [comment]
Per https://github.com/angular/angular-cli/issues/9016, adding the -prod flag includes a lot of compiler options that help shrink the binary:

Uncompressed:
-rw-r--r-- 1 rachelrosen primarygroup 1.5K Apr 19 10:35 inline.b6085dbd30b7d451e456.bundle.js
-rw-r--r-- 1 rachelrosen primarygroup 1.4M Apr 19 10:35 main.c32fb536f5ea1d697589.bundle.js
-rw-r--r-- 1 rachelrosen primarygroup  83K Apr 19 10:35 polyfills.f2547cc69b1ca9b87c7d.bundle.js

= 1.48 MB

gzipped:
-rw-r--r-- 1 rachelrosen primarygroup  843 Apr 19 10:35 inline.b6085dbd30b7d451e456.bundle.js.gz
-rw-r--r-- 1 rachelrosen primarygroup 365K Apr 19 10:35 main.c32fb536f5ea1d697589.bundle.js.gz
-rw-r--r-- 1 rachelrosen primarygroup  28K Apr 19 10:35 polyfills.f2547cc69b1ca9b87c7d.bundle.js.gz

= ~393 K

Additionally using the -vc flag:
-rw-r--r-- 1 rachelrosen primarygroup 1.5K Apr 19 10:40 inline.95f2ecf746b368ba5e01.bundle.js
-rw-r--r-- 1 rachelrosen primarygroup 178K Apr 19 10:40 main.016367d69f199d542b91.bundle.js
-rw-r--r-- 1 rachelrosen primarygroup  83K Apr 19 10:40 polyfills.3ffba3192cb00ea552ef.bundle.js
-rw-r--r-- 1 rachelrosen primarygroup 1.2M Apr 19 10:40 vendor.9001737a910243c4e0f4.bundle.js

= 1.46 M

With gzip:
-rw-r--r-- 1 rachelrosen primarygroup  844 Apr 19 10:40 inline.95f2ecf746b368ba5e01.bundle.js.gz
-rw-r--r-- 1 rachelrosen primarygroup  25K Apr 19 10:40 main.016367d69f199d542b91.bundle.js.gz
-rw-r--r-- 1 rachelrosen primarygroup  28K Apr 19 10:40 polyfills.3ffba3192cb00ea552ef.bundle.js.gz
-rw-r--r-- 1 rachelrosen primarygroup 348K Apr 19 10:40 vendor.9001737a910243c4e0f4.bundle.js.gz

= ~401 K

**Comment by rachelrosengoogle** [comment]
Tl;dr, turns out the error was just not including the -prod flag in the prod build. =)

**Comment by rachelrosengoogle** [comment]
Turns out It can be even smaller by not using the -aot option:

npm run setup && ng build -prod --env=prod -aot=false --build-optimizer=true --extract-css=false

-rw-r--r-- 1 rachelrosen primarygroup 1.5K Apr 19 17:49 inline.8e40830d8ae54c554683.bundle.js
-rw-r--r-- 1 rachelrosen primarygroup 873K Apr 19 17:49 main.df4074be3ba378a6232e.bundle.js
-rw-r--r-- 1 rachelrosen primarygroup  84K Apr 19 17:49 polyfills.101d31ca0755097952b5.bundle.js
-rw-r--r-- 1 rachelrosen primarygroup  48K Apr 19 17:49 styles.aa439def96e82eb02b45.bundle.js

= ~1MB

With gzip:
-rw-r--r-- 1 rachelrosen primarygroup  845 Apr 19 17:49 inline.8e40830d8ae54c554683.bundle.js.gz
-rw-r--r-- 1 rachelrosen primarygroup 240K Apr 19 17:49 main.df4074be3ba378a6232e.bundle.js.gz
-rw-r--r-- 1 rachelrosen primarygroup  28K Apr 19 17:49 polyfills.101d31ca0755097952b5.bundle.js.gz
-rw-r--r-- 1 rachelrosen primarygroup 8.3K Apr 19 17:49 styles.aa439def96e82eb02b45.bundle.js.gz

=~276K
